react-form-controls/src/index.jsx

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+class NewsletterForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { email: '' };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(event) {
+    this.setState({ email: event.target.value });
+  }
+
+  handleSubmit(event) {
+    console.log('state: ', this.state);
+    event.preventDefault();
+  }
+
+  render() {
+    return (
+      <form onChange={this.handleChange} onSubmit={this.handleSubmit}>
+        <label htmlFor='email'>Email: </label>
+        <input type='email' name='email' id='email'></input>
```

Comment:
The `onChange` handler should be on the input, also the input should have a value prop set from state.